activestorage/app/javascript/activestorage/ujs.js

```diff
@@ -58,8 +58,8 @@ function submitForm(form) {
     button.disabled = disabled
   } else {
     button = document.createElement("input")
-    button.type = "submit"
-    button.style = "display:none"
+    button.setAttribute("type", "submit")
```

Comment:
Is this change required too? Looking at #32378 and other search results for "Assignment to read-only properties is not allowed in strict mode" it seems like the issue is just with setting `button.style = "…"`.

Reply:
From my testing it required both to be changed. Once I changed the style attribute I got the same error on the way that the type was set to submit. Making these two changes then allowed it to work properly.

Reply:
I spelled that out on a comment below but I’m happy to update the issue description if that helps as well.

Reply:
Hmm, I'm only seeing an error in IE 11 setting `button.style = "…"` (in strict mode only). Tested with:
```js
(function() {
  "use strict";
  
  var button = document.createElement("button")
  
  try {
    button.type = "submit"
  } catch (error) {
    console.log("error setting button.type", error)
  }

  try {
    button.style = "display:none"
  } catch (error) {
    console.log("error setting button.style", error)
  }

  try {
    button.style.display = "none"
  } catch (error) {
    console.log("error setting button.style.display", error)
  }
})()
```

Reply:
You're correct - when I first began testing I wasn't sure how to run off of a local rails gem in my gemfile, so when I ran my testing I may not have properly testing against just the style change.

I've just confirmed on my side that simply changing the style configuration step solves the problem.

Sorry for the confusion on this!

Reply:
Would you like me to submit a new PR? Happy to do so if you'd find it useful.